video/out/cocoa_cb_common.swift

```diff
@@ -122,7 +122,41 @@ class CocoaCB: Common {
 
         libmpv.setRenderICCProfile(colorSpace)
         if #available(macOS 10.11, *) {
-            layer?.colorspace = colorSpace.cgColorSpace
+
+            // Experimental #HDR on #macOS        
+            var name = "" as CFString
+
+            // HDR is supported fully only on macOS 10.15 and higher
+#if HAVE_MACOS_10_15_FEATURES
+                switch (libmpv.macOpts.macos_hdr_csp)
+                {
+                    case 0: name = CGColorSpace.displayP3_HLG
+                            break;
+                    case 1: name = CGColorSpace.displayP3_PQ_EOTF
+                            break;
+                    case 2: name = CGColorSpace.extendedLinearDisplayP3
+                            break;
+                    case 3: name = CGColorSpace.itur_2020
+                            break;
+                    case 4: name = "" as CFString 
+                            if #available(macOS 10.15.6, *) {
+                                name = CGColorSpace.itur_2020_HLG 
+                            }
+                            break;
+                    case 5: name = CGColorSpace.itur_2020_PQ_EOTF
+                            break;
+                    default: 
+                            break;
+                }
+#endif            
+            // Experimental #HDR on #macOS        
+            if (name as String != "")
+            {
+                layer?.colorspace = CGColorSpace(name: name)
+            } else
```

Comment:
i am not a big fan of the if else here. i think it would be the best to create a variable at the beginning with the auto/default colorSpace, overwrite it with the switch case and set this variable to the layer's color space.